docs/kb/semgrep-ci/azure-self-hosted-ubuntu.md

```diff
@@ -52,15 +52,15 @@ steps:
   persistCredentials: true
 - script: |
     pipx install semgrep
-    if [ $(Build.SourceBranchName) = "master" ]; then
-        echo "Semgrep full scan"
-        semgrep ci
-    elif [ $(System.PullRequest.PullRequestId) -ge 0 ]; then
+    if [ $(System.PullRequest.PullRequestId) -ge 0 ]; then
```

Comment:
@armchairlinguist I inverted this section to eliminate the need for hardcoding the default branch name. Is this too strange/not done? WDYT? cc: @stuartcmehrens

Reply:
What you've done isn't equivalent. It may be something we want to do, but we should discuss the approach.

This config runs full scans on the default branch (unfortunately hardcoded; I'm not sure if we can detect it) and diff scans on pull requests. Your config runs full scans on any branch unless a pull request ID is provided. This could mean that we would run a full scan on a feature branch before a PR is opened, and therefore mess with PR comments. I wouldn't recommend making the change as-is for that reason.

Reply:
Got it; thanks for the sanity check! I'm undoing all of this and will revisit at a later date after we talk about the best way to do so.